package/MDAnalysis/coordinates/PDB.py

```diff
@@ -272,41 +273,34 @@ def __init__(self, filename, **kwargs):
         models = []
         crysts = []
 
-        pdbfile = self._pdbfile = util.anyopen(filename)
+        # hack for streamIO
+        if isinstance(filename, util.NamedStream) and isinstance(filename.stream, cStringIO):
+            filename.stream = io.BytesIO(filename.stream.getvalue().encode())
+
+        pdbfile = self._pdbfile = util.anyopen(filename, 'rb')
```

Comment:
[`anyopen` docs](https://www.mdanalysis.org/docs/documentation_pages/lib/util.html?highlight=anyopen#MDAnalysis.lib.util.anyopen) indicate that

> only the first letter of `mode` is used and thus any additional modifiers are silently ignored

I took a quick look at the source and we do seem to pass through the `mode`, but there's a bunch of handling code in there that may not really use it after all--are we sure this is using binary properly?